Post/views.py

```diff
@@ -110,3 +111,22 @@ def get_post_by_query_text(query_text):
     return Post.objects.filter(
         nameOfLocation__icontains=query_text
     ) | Post.objects.filter(Description__icontains=query_text)
+
+
+class PostUpdateView(
+    LoginRequiredMixin, UserPassesTestMixin, SuccessMessageMixin, UpdateView
+):
+    model = Post
+    template_name = 'Post/updatePost.html'
+    fields = ['nameOfLocation', 'photoURL', 'Description']
+    success_message = 'The post has been successfully updated!'
+
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        return super().form_valid(form)
+
+    def test_func(self):
+        post = self.get_object()
+        if self.request.user == post.user:
+            return True
+        return False
```

Comment:
assert..

Reply:
@Shoval-Ovadia why not using an assert here?

Reply:
@Yarboa @guy9050  I tried - it is not working that way.
I changed the function that way:
![image](https://user-images.githubusercontent.com/79395387/118955928-d28a9680-b967-11eb-9d88-d1f28befc312.png)

And when I tried to update a post that I created - I get a 403 error (so this function returns 'false').
Should I do something else?

Reply:
So, i think it should be as following
obj = self.get_object()
return obj.author == self.request.user

Reply:
Works!